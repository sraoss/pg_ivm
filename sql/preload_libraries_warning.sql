-- SQL tests to validate that the session_preload_libraries warning is emitted
-- whenever pg_ivm is not in session_preload_libraries.

DROP EXTENSION IF EXISTS pg_ivm CASCADE;

-- Validate that pg_ivm 1.10 cannot be created if pg_ivm is not in
-- session_preload_libraries.
ALTER SYSTEM RESET session_preload_libraries;
SELECT pg_reload_conf();
\c -

CREATE EXTENSION pg_ivm VERSION '1.9';  -- expect success
ALTER EXTENSION pg_ivm UPDATE;          -- expect failure
DROP EXTENSION pg_ivm;
CREATE EXTENSION pg_ivm;                -- expect failure

-- Validate that pg_ivm 1.10 can be created if pg_ivm is in
-- session_preload_libraries.
ALTER SYSTEM SET session_preload_libraries = pg_ivm;
SELECT pg_reload_conf();
\c -

CREATE EXTENSION pg_ivm VERSION '1.9';  -- expect success
ALTER EXTENSION pg_ivm UPDATE;          -- expect success
DROP EXTENSION pg_ivm;
CREATE EXTENSION pg_ivm;                -- expect success

DROP EXTENSION pg_ivm CASCADE;
