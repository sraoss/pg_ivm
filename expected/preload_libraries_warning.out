-- SQL tests to validate that the session_preload_libraries warning is emitted
-- whenever pg_ivm is not in session_preload_libraries.
DROP EXTENSION IF EXISTS pg_ivm CASCADE;
NOTICE:  drop cascades to 12 other objects
DETAIL:  drop cascades to trigger IVM_trigger_ins_before_17633 on table t
drop cascades to trigger IVM_trigger_del_before_17634 on table t
drop cascades to trigger IVM_trigger_upd_before_17635 on table t
drop cascades to trigger IVM_trigger_truncate_before_17636 on table t
drop cascades to trigger IVM_trigger_ins_after_17637 on table t
drop cascades to trigger IVM_trigger_del_after_17638 on table t
drop cascades to trigger IVM_trigger_upd_after_17639 on table t
drop cascades to trigger IVM_trigger_truncate_after_17640 on table t
drop cascades to trigger IVM_prevent_immv_change_17621 on table mv
drop cascades to trigger IVM_prevent_immv_change_17622 on table mv
drop cascades to trigger IVM_prevent_immv_change_17623 on table mv
drop cascades to trigger IVM_prevent_immv_change_17624 on table mv
-- Validate that pg_ivm 1.10 cannot be created if pg_ivm is not in
-- session_preload_libraries.
ALTER SYSTEM RESET session_preload_libraries;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

\c -
CREATE EXTENSION pg_ivm VERSION '1.9';  -- expect success
ALTER EXTENSION pg_ivm UPDATE;          -- expect failure
ERROR:  pg_ivm is not loaded in shared_preload_libraries or session_preload_libraries
HINT:  Add pg_ivm to session_preload_libraries and restart the session. Or, add pg_ivm to shared_preload_libraries and restart Postgres.
CONTEXT:  PL/pgSQL function inline_code_block line 13 at RAISE
DROP EXTENSION pg_ivm;
CREATE EXTENSION pg_ivm;                -- expect failure
ERROR:  pg_ivm is not loaded in shared_preload_libraries or session_preload_libraries
HINT:  Add pg_ivm to session_preload_libraries and restart the session. Or, add pg_ivm to shared_preload_libraries and restart Postgres.
CONTEXT:  PL/pgSQL function inline_code_block line 13 at RAISE
-- Validate that pg_ivm 1.10 can be created if pg_ivm is in
-- session_preload_libraries.
ALTER SYSTEM SET session_preload_libraries = pg_ivm;
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

\c -
CREATE EXTENSION pg_ivm VERSION '1.9';  -- expect success
ALTER EXTENSION pg_ivm UPDATE;          -- expect success
DROP EXTENSION pg_ivm;
CREATE EXTENSION pg_ivm;                -- expect success
DROP EXTENSION pg_ivm CASCADE;
